src/AWSMetadataUtilities.jl

```diff
@@ -0,0 +1,578 @@
+module AWSMetadataUtilities
+
+include("AWSExceptions.jl")
+
+using .AWSExceptions
+using OrderedCollections: OrderedDict, LittleDict
+using HTTP
+using JSON
+using Mocking
+
+
+"""
+    _get_aws_sdk_js_files()
+
+Get a list of all AWS service API definition files from the `awsk-sdk-js` GitHub repository.
+
+# Returns
+- `Array{OrderedDict}`: Array of Dictionaries which contains information about each AWS Service
+"""
+function _get_aws_sdk_js_files()
+    headers = ["User-Agent" => "JuliaCloud/AWS.jl"]
+    url = "https://api.github.com/repos/aws/aws-sdk-js/contents/apis"
+
+    req = @mock HTTP.get(url, headers)
+    files = JSON.parse(String(req.body), dicttype=OrderedDict)
+    filter!(f -> endswith(f["name"], ".normal.json"), files)  # Only get ${Service}.normal.json files
+    files = _filter_latest_service_version(files)
+
+    return files
+end
+
+"""
+    _filter_latest_service_version(services::Array{Any})
+
+Return a list of all AWS Services and their latest version.
+
+# Arguments
+- `services::Array`: Array of all AWS APIs and their versions
+
+# Returns
+- `Array{Dict}`: Array of the latest AWS Service definitions
+"""
+function _filter_latest_service_version(services::Array)
```

Comment:
Also update docstring
```suggestion
function _filter_latest_service_version(services::AbstractArray)
```